API-Samples/init_texture/init_texture.cpp

```diff
@@ -83,8 +83,8 @@ int sample_main(int argc, char *argv[]) {
     vkGetPhysicalDeviceFormatProperties(info.gpus[0], VK_FORMAT_R8G8B8A8_UNORM, &formatProps);
 
     /* See if we can use a linear tiled image for a texture, if not, we will
-     * need a staging image for the texture data */
-    bool needStaging = (!(formatProps.linearTilingFeatures & VK_FORMAT_FEATURE_SAMPLED_IMAGE_BIT)) ? true : false;
+     * need a staging buffer for the texture data */
+    texObj.needs_staging = (!(formatProps.linearTilingFeatures & VK_FORMAT_FEATURE_SAMPLED_IMAGE_BIT)) ? true : false;
```

Comment:
if the type is bool no need for a ```? true : false``` ternary in C++, the logical expression directly evaluates to the bool.